tensorflow_datasets/core/dataset_builder.py

```diff
@@ -365,13 +400,13 @@ def download_and_prepare(self, download_dir=None, download_config=None):
 
   @api_utils.disallow_positional_args
   def as_dataset(self,
-                 split=None,
-                 batch_size=None,
-                 shuffle_files=False,
-                 decoders=None,
-                 read_config=None,
-                 as_supervised=False,
-                 in_memory=None):
+                 split: splits_lib.SplitBase = None,
```

Comment:
`Optional[splits_lib.SplitBase]` here and bellow

Reply:
Done